src/main/java/com/google/graphgeckos/dashboard/components/BuildBotData.java

```diff
@@ -0,0 +1,146 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.components;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.cloud.Timestamp;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+import org.springframework.data.annotation.Transient;
+
+/**
+ * Contains the information retrieved from a single build bot. It is used as a member of BuildInfo...
+ * of {@link BuildInfo}
+ */
+@Entity(name = "builder")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class BuildBotData {
+
+  @Transient
+  private String commitHash;
+
+  /* The timestamp of the build. */
+  private Timestamp timestamp;
+
+  /* Name of the buildbot. */
+  private final String name;
+
+  /**
+   * The logs of each compilation stage, stored as described by {@link Log}.
+   */
+  private final List<Log> logs = new ArrayList<>();
+
+  /**
+   * Builder compilation status, as described by {@link BuilderStatus}.
+   */
+  private BuilderStatus status;
+
+  /**
+   * Extracts nested commitHash("revision"), branch("branch") and timestamp("when")
+   * fields from the json.
+   * @param sourceStamp Representation of the json component, where the commitHash field is located.
+   */
+  @JsonProperty("sourceStamp")
+  public void unpackSourceStamp(Map<String, Object> sourceStamp) {
+    commitHash = sourceStamp.get("revision").toString();
+    timestamp = Timestamp.of(new Date(sourceStamp.get("when").toString()));
+  }
+
+  /**
+   * Defines builder status based on a phrase provided in parsed json.
+   * If something failed, then the buidbot is considered as failed {@code FAILED}.
+   * If something lost, then the buildbot is considered lost {@code LOST}.
+   * Otherwise the buildbot is considered passed {@code PASSED}.
+   * @param words Words of the "text" json field.
+   */
+  @JsonProperty("text")
+  private void extractStatus(List<String> words) {
+    for (String word : words) {
+      if (word.equals("failed")) {
+        status = BuilderStatus.FAILED;
+        break;
+      }
+      if (word.equals("lost")) {
+        status = BuilderStatus.LOST;
+        break;
+      }
+      if (word.equals("successful")) {
+        status = BuilderStatus.PASSED;
+      }
+    }
+  }
+
+  /**
+   * Unpacks logs represented as list of lists of two strings,
+   * where the first one is a type of the log and the second one
+   * is a link to the log.
+   * @param logs Representation of the json component, where the logs are located.
+   */
+  @JsonProperty("logs")
+  private void unpackLogs(List<String[]> logs) {
+    logs.forEach(x -> this.logs.add(new Log(x)));
+  }
+
+  public BuildBotData(@JsonProperty("builderName") String name) {
```

Comment:
I might not understand something regarding this. Is the annotation `@JsonProperty("builderName")` inside the method's parameters a mistake? If not, why isn't it like that for all methods?

Reply:
no

Reply:
why?